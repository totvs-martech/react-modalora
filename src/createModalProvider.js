import React from 'react'
import ModalProviderOriginal from './ModalProvider'

const createModalProvider = ({ ModalProvider = ModalProviderOriginal, ...opts } = {}) =>
  props => <ModalProvider {...opts} {...props} />

export default createModalProvider
