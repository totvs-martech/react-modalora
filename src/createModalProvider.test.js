import React from 'react'
import { render } from 'react-testing-library'
import createModalProvider from './createModalProvider'

describe('createModalProvider', function () {
  it('should create an ModalProvider and match snapshot', function () {
    const ModalProvider = createModalProvider()
    render(<ModalProvider>{() => 'foo'}</ModalProvider>)
    expect(document.body.innerHTML).toMatchSnapshot()
  })

  it('should pass custom props', function () {
    const content = () => 'bar'
    const ModalProvider = createModalProvider({
      content
    })
    render(<ModalProvider>{() => 'foo'}</ModalProvider>)
    expect(document.body.innerHTML).toMatchSnapshot()
  })

  it('should be able to optionaly close the modal on overlay click', function () {
    const content = () => 'bar'
    const ModalProvider = createModalProvider({
      content,
      closeOnOverlayClick: true
    })
    render(<ModalProvider>{() => 'foo'}</ModalProvider>)
    expect(document.body.innerHTML).toMatchSnapshot()
  })
})
