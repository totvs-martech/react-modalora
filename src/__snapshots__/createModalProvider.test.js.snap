// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createModalProvider should create an ModalProvider and match snapshot 1`] = `

<div>
  foo
</div>

`;

exports[`createModalProvider should pass custom props 1`] = `

<div>
  foo
</div>
<div>
  <div style="align-items: flex-start; bottom: 0px; display: flex; justify-content: center; left: 0px; max-height: 100vh; max-width: 100vw; overflow: auto; position: fixed; right: 0px; top: 0px;">
    <div style="background-color: rgba(0, 0, 0, 0.2); bottom: 0px; left: 0px; position: fixed; right: 0px; top: 0px; transform: translate3d(0px, -1px, 0px);">
    </div>
    <div role="dialog"
         style="background-color: rgb(255, 255, 255); margin: 3rem auto; max-width: 100vw; padding: 1.5rem; transform: translate3d(0px, 1px, 0px); width: 80rem;"
    >
      bar
    </div>
  </div>
</div>

`;
